### ISSUE TITLE
Deprecation log warning about <dateAsTimestamp/> should link to relevant manual section

### ISSUE THREAD

**Issue by akoppathn** [jOOQ/jOOQ#15843]
### Your question

I am attempting to replicate the same behavior as the deprecated `dateAsTimestamp` flag (Oracle `DATE` columns translated to `LocalDateTime`). I've read through the documentation several times and have also perused the various support forums for past discussion on this (including #12798). Unfortunately, I am still thoroughly confused. Is there any chance of getting a section added to the `dateAsTimestamp` page that provides the code needed to migrate away from that flag?

### jOOQ Version

jOOQ Professional 3.18

### Database product and version

Oracle 19c

**Comment by lukaseder** [comment]
Thanks for your suggestion. Sure, we can add a paragraph. What are you confused with, specifically? How to implement a `Binding`? Or how to attach it to generated code?

**Comment by akoppathn** [comment]
I guess I am a bit unsure about what needs to happen conceptually.

I think I need to create a forced type that includes types of `DATE` and a user type of `LocalDateTime`. When I compare the code generated with `dateAsTimestamp` enabled with the code generated by this, it still uses `SQLDataType.LOCALDATE` in the "type" parameter, whereas `dateAsTimestamp` ends up using `SQLDataType.LOCALDATETIME(0)`. I can replicate that behavior by adding a `name` to the forced type of `TIMESTAMP`. At that point, the only difference is the usage of the `LocalDateAsLocalDateTimeBinding` binding. Unfortunately, when I attempt to modify the forced type to include that binding, it doesn't seem to actually apply it.

As an aside, the output indicates that the forced type is not being used. It clearly is, so I am not sure what's up with that.

Based on other responses, I'm beginning to suspect attempting to replicate the exact code generated by `dateAsTimestamp` via forced types is not recommended for one reason or another.

**Comment by akoppathn** [comment]
Here's the relevant piece. 
```
                            .withForcedTypes(
                                new ForcedType()
                                    .withName("TIMESTAMP")
                                    .withUserType(LocalDateTime.class.getCanonicalName())
                                    .withObjectType(ForcedTypeObjectType.COLUMN)
                                    .withIncludeTypes("DATE")
                                    .withBinding(
                                        LocalDateAsLocalDateTimeBinding.class.getCanonicalName())))
```

Although I now see the warning that seems to indicate why the binding isn't getting applied:

`Bad configuration for <forcedType/>. <userType/> is not allowed when <name/> is provided: <priority>0</priority><name>TIMESTAMP</name><userType>java.time.LocalDateTime</userType><binding>org.jooq.impl`

`Bad configuration for <forcedType/>. <binding/> is not allowed when <name/> is provided: <priority>0</priority><name>TIMESTAMP</name><binding>org.jooq.impl.LocalDateAsLocalDateTimeBinding</binding><includeTypes>DATE</includeTypes><nullability>ALL</nullability><objectType>COLUMN</objectType>`

**Comment by lukaseder** [comment]
Yes, the `name` property is used for data type rewrites, i.e. when you want jOOQ to simply replace `DATE` by `TIMESTAMP` in the meta data input:
https://www.jooq.org/doc/dev/manual/code-generation/codegen-advanced/codegen-config-database/codegen-database-forced-types/codegen-database-forced-types-rewrite/

But then, all information about the original type is lost, so in your case, just remove that.

**Comment by akoppathn** [comment]
Yeah, but removing the `withName` results in `SQLDataType.LOCALDATE` being used in `createField` which seems to be incompatible with the pre-existing `LocalDateAsLocalDateTimeBinding` binding. Would it be a fair conclusion then that the `LocalDateAsLocalDateTimeBinding` can't actually be used in this situation?

**Comment by lukaseder** [comment]
> Yeah, but removing the `withName` results in `SQLDataType.LOCALDATE` being used in `createField` which seems to be incompatible with the pre-existing `LocalDateAsLocalDateTimeBinding` binding

Can you show the generated code that you're getting?

**Comment by akoppathn** [comment]
`public final TableField<MytableRecord, LocalDateTime> CREATED = createField(DSL.name("CREATED"), SQLDataType.LOCALDATE, this, "", new LocalDateAsLocalDateTimeBinding());`

Which, the IDE indicates: `Cannot resolve method 'createField(Name, DataType<LocalDate>, Mytable, String, LocalDateAsLocalDateTimeBinding)'`

**Comment by lukaseder** [comment]
Ah, I see it now, I'm sorry for the confusion on my side - it's been a while with this change. And indeed, I can see now why this doesn't compile.

The `LocalDateAsLocalDateTimeBinding` implementation has been added as a retrofitting for the `<dateAsTimestamp>` implementation. It seems that the `name` property would be necessary to make this work, but the `GenerationTool` forbids it.

It seems that this feature request would help here:

- https://github.com/jOOQ/jOOQ/issues/13295

Will have a closer look at this soon.

**Comment by lukaseder** [comment]
https://github.com/jOOQ/jOOQ/issues/13295 has been implemented for jOOQ 3.19.0. The change is a bit delicate as it touches other deprecated functionality (usage of `<customType>` declarations), so a backport would risk breaking things.

The `<dateAsTimestamp>` flag won't be removed in the next releases, so fixing your code generation configuration isn't urgent. Starting from jOOQ 3.19, you will be able to combine `<name>` with `<binding>` (which is what `<dateAsTimestamp>` does, behind the scenes), in case of which the compilation error can be fixed.
